api/core/Directus/Filesystem/Thumbnailer.php

```diff
@@ -219,12 +219,12 @@ public function extractThumbnailParams($thumbnailUrlPath)
             
             // make sure filename is valid
             $info = pathinfo($fileName);
-            if (! in_array(ArrayUtils::get($info, 'extension'), $this->getSupportedFileExtensions())) {
+            if (! $this->isSupportedFileExtension((ArrayUtils::get($info, 'extension')))) {
                 throw new Exception('Invalid file extension.');
             }
             
             $thumbnailParams = [
-                'fileName' => $fileName,
+                'fileName' => ArrayUtils::get($info, 'filename') . '.' . strtolower(ArrayUtils::get($info, 'extension')),
```

Comment:
Shouldn't only the "checker" (the method that validates the extension) be the only case sensitive, instead of convert the extension from the original to lowercase? or is there a reason to do this?

Reply:
Yes, both need to normalize it.  The validator is checking against lowercase file type strings and the 'getter' is looking for lowercase file extensions.

Reply:
supporting `.jpg` mens supporting `.JPG` and `.jpg` those are the same file extension, but we can have `FILE1.JPG` and `file1.jpg` which can be different, but it was just something to think about it. because the issue was checking whether it was supported not how the actually fixed was going to be generated.

✌️

Reply:
How is the Directus file generated?  Isn't it normalized to '0000001.jpg', for example?

Reply:
Late reply, but it uses the original name, that was were you confusion came from probably. You can read the issue was referring being reported here: https://github.com/directus/directus/issues/1989

I will take care of this one.